Tasks/AzurePowerShellV4/TryMakingModuleAvailable.ps1

```diff
@@ -0,0 +1,78 @@
+[CmdletBinding()]
+param (
+    [string]
+    $targetVersion,
+
+    [string] [Parameter(Mandatory = $true)] [ValidateSet("Windows", "Linux", "Mac")]
+    $platform
+)
+
+try {
+    $isWin = $platform -eq "Windows"
+    if ($isWin) {
+        . "$PSScriptRoot\Utility.ps1"
+        $isHostedAgent = Test-IsHostedAgentPathPresent
+    } else {
+        . "$PSScriptRoot/Utility.ps1"
+        $isHostedAgent = Test-IsHostedAgentPathPresentLinux
+    }
+
+    if (!$isHostedAgent) {
+        $moduleSource = "Others"
+        return
+    }
+
+    if (!$targetVersion) {
```

Comment:
if targetVersion being null causes the function to return, why not do it as the first check in this method ?

Reply:
We want to have a check for hostedAgent first so that we can mark moduleSource as Others in that case even if targetVersion is "".
If we make this check in the beginning the we would end up logging moduleSource as Folder which don't expect.